watertap/core/zero_order_unit.py

```diff
@@ -0,0 +1,264 @@
+            default=tmp_dict_2)
+
+        # Create Ports
+        self.add_port("inlet", self.properties_in, doc="Inlet port")
+        self.add_port("outlet", self.properties_out, doc="Outlet port")
+        self.add_port("waste", self.properties_waste, doc="Waste port")
+
+        # Add performance variables
+        self.water_recovery = Var(
+            self.flowsheet().time,
+            initialize=0.8,
+            domain=NonNegativeReals,
+            units=pyunits.dimensionless,
+            bounds=(1E-8, 1.0000001),
+            doc='Water recovery fraction')
+        self.removal_fraction = Var(
+            self.flowsheet().time,
+            self.config.property_package.solute_set,
+            domain=NonNegativeReals,
+            initialize=0.01,
+            units=pyunits.dimensionless,
+            doc='Solute removal fraction')
+
+        self.deltaP_outlet = Var(
+            self.flowsheet().time,
+            initialize=0,
+            units=units_meta('pressure'),
+            doc='Pressure change between inlet and outlet')
+        self.deltaP_waste = Var(
+            self.flowsheet().time,
+            initialize=0,
+            units=units_meta('pressure'),
+            doc='Pressure change between inlet and waste')
+
+        # Add performance constraints
+        # Water recovery
+        @self.Constraint(self.flowsheet().time, doc='Water recovery equation')
+        def water_recovery_equation(b, t):
+            return (b.water_recovery[t] * b.properties_in[t].flow_vol ==
+                    b.properties_out[t].flow_vol)
+
+        # Flow balance
+        @self.Constraint(self.flowsheet().time, doc='Overall flow balance')
+        def flow_balance(b, t):
+            return (b.properties_in[t].flow_vol ==
+                    b.properties_out[t].flow_vol +
+                    b.properties_waste[t].flow_vol)
+
+        # Solute removal
+        @self.Constraint(self.flowsheet().time,
+                         self.config.property_package.solute_set,
+                         doc='Solute removal equations')
+        def solute_removal_equation(b, t, j):
+            return (b.removal_fraction[t, j] *
+                    b.properties_in[t].get_material_flow_terms("Liq", j) ==
+                    b.properties_waste[t].get_material_flow_terms("Liq", j))
+
+        # Solute balances
+        @self.Constraint(self.flowsheet().time,
+                         self.config.property_package.solute_set,
+                         doc='Solute mass balance equations')
+        def solute_mass_balances(b, t, j):
+            return (b.properties_in[t].get_material_flow_terms("Liq", j) ==
+                    b.properties_out[t].get_material_flow_terms("Liq", j) +
+                    b.properties_waste[t].get_material_flow_terms("Liq", j))
+
+        # Pressure drop
+        @self.Constraint(self.flowsheet().time, doc='Outlet pressure equation')
+        def outlet_pressure_constraint(b, t):
```

Comment:
For the case of NF/RO, I don't think we would need this.

Reply:
This ties into the comment above on whether and how `deltaP` should be an optional term in these types of units.